### PR TITLE
Make `skip_history_when_saving` work when creating a new object

### LINKED ISSUE #1141: skip_history_when_saving does not work as expected
**Describe the bug**
I am trying to avoid writing to history when certain condition is met. I have a base model that looks the following.
```python
class BaseModel(models.Model):
    ...
    history = HistoricalRecords(inherit=True)
    
    def save(self, *, skip_history_when_saving=True, **kwargs):
        if skip_history_when_saving:
            self.skip_history_when_saving = skip_history_when_saving
        super().save(**kwargs)
```
For my models, I am creating it using:
```python
class MyModel(BaseModel):
    ...

my_model = MyModel()
my_model.full_clean()
my_models.save(skip_history_when_saving=True) #expecting only one insertion into the MyModel db but seeing an entry being created for simple history MyModel table as well)
```
`save()` gets called as expected in the base model, at the end of execution, it's still creating an entry in historical table. I can confirm it by looking at the sql query. 

**Expected behavior**
Expecting only the model instance to be created without creating the historical model instance on post creation. 

**Environment (please complete the following information):**
 - Django Simple History Version: [e.g. 3.2.0]
 - Django Version: [e.g. 4.1.7]

### PR DESCRIPTION
**Pull request by ddabble**
## Description

`skip_history_when_saving` used to only apply when updating an existing object (due to [the `not created` operand in `post_save()`](https://github.com/jazzband/django-simple-history/commit/3c2319c4f9164fb51152e94a03c27f2d978cae9e#diff-a13c48090526a77430e6819c61ffb009b8ed15ffca1b40f3eef675c8c9b77db9L654)), not when creating a new one. This PR fixes that.

Also improved the docs on saving without creating historical records.

## Related Issue

Closes #1141.

## Motivation and Context

Fixing the linked issue.

## How Has This Been Tested?

See the test added in `test_models.py`.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have run the `pre-commit run` command to format and lint.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] I have added my name and/or github handle to `AUTHORS.rst`
- [x] I have added my change to `CHANGES.rst`
- [x] All new and existing tests passed.

### CODE REVIEW

File: docs/querying_history.rst [hunk 1]
```diff
@@ -175,30 +175,49 @@ model history.
     <Poll: Poll object as of 2010-10-25 18:04:13.814128>
 
 
-Save without a historical record
---------------------------------
+Save Without Creating Historical Records
```
Comment (tim-schilling) [inline docs/querying_history.rst:178]: Most titles don't use title casing in the file. I think you should reconsider this choice.

File: docs/querying_history.rst [hunk 2]
```diff
@@ -175,30 +175,49 @@ model history.
...
+----------------------------------------
 
-If you want to save a model without a historical record, you can use the following:
+If you want to save model objects without triggering the creation of any historical
+records, you can set a ``skip_history_when_saving`` attribute to ``True``
+on each object before saving - for example like this:
```
Comment (tim-schilling) [inline docs/querying_history.rst:183]: Personally, I think the previous version is better. It's fewer words and you have a code example to show the how.

File: docs/querying_history.rst [hunk 3]
```diff
@@ -175,30 +175,49 @@ model history.
...
 
-        def save_without_historical_record(self, *args, **kwargs):
-            self.skip_history_when_saving = True
-            try:
-                ret = self.save(*args, **kwargs)
-            finally:
-                del self.skip_history_when_saving
-            return ret
+This also works when creating an object
+(albeit only when calling the model's standard Python constructor followed by saving):
```
Comment (tim-schilling) [inline docs/querying_history.rst:193]:
```suggestion
This also works when creating an object, but only when calling the ``save`` method:
```

I think it simplifies the language a bit. I'm not tied particularly strongly to it.

File: docs/querying_history.rst [hunk 4]
```diff
@@ -175,30 +175,49 @@ model history.
...
+
+.. code-block:: python
 
+    # Note that `Poll.objects.create()` is not called
```
Comment (tim-schilling) [inline docs/querying_history.rst:197]: Maybe we put this after the code example?

> Historical records will always be created using the ``create`` manager method.

File: docs/querying_history.rst [hunk 5]
```diff
@@ -175,30 +175,49 @@ model history.
...
+    def save_without_historical_record(self, *args, **kwargs):
+        self.skip_history_when_saving = True
+        try:
+            ret = self.save(*args, **kwargs)
+        finally:
+            del self.skip_history_when_saving
+        return ret
 
-Or disable history records for all models by putting following lines in your ``settings.py`` file:
+As a final option, you can disable the creation of historical records for *all* models
```
Comment (tim-schilling) [inline docs/querying_history.rst:219]:
```suggestion
Or disable the creation of historical records for *all* models
```

I think the previous version was a bit too wordy for technical documentation.

File: simple_history/models.py [hunk 6]
```diff
@@ -180,9 +180,9 @@ def contribute_to_class(self, cls, name):
     def add_extra_methods(self, cls):
         def save_without_historical_record(self, *args, **kwargs):
             """
-            Save model without saving a historical record
+            Save the model instance without creating a historical record.
 
-            Make sure you know what you're doing before you use this method.
+            *Make sure you know what you're doing before using this method.*
```
Comment (tim-schilling) [inline simple_history/models.py:185]: nit: I'm not a fan of trying to use markup here.

File: docs/querying_history.rst (hunk 4 above)
Comment (tim-schilling) [reply docs/querying_history.rst:197]: Also I bet it's possible for us to manage this by overriding the ``create`` method on the manager class and accepting ``skip_history_when_saving`` as a kwarg.

**Review by tim-schilling — CHANGES_REQUESTED** [review]
If you disagree with my review, I could be convinced to leave things as is.

File: docs/querying_history.rst (hunk 1 above)
Comment (ddabble) [reply docs/querying_history.rst:178]: Ah yes, I agree 👌 I was planning on polishing the docs in a later PR, which includes title-casing all the headings - to match both the main headings in the docs sidebar and various subheadings on some of the pages.

Any opinions on whether we should make title casing or normal casing the norm for the subheadings, btw?

File: docs/querying_history.rst (hunk 2 above)
Comment (ddabble) [reply docs/querying_history.rst:183]: Good point on already having a code example 😅 I'll push a slightly shorter version 🙂 I do prefer being specific about "triggering the creation", though.

File: docs/querying_history.rst (hunk 3 above)
Comment (ddabble) [reply docs/querying_history.rst:193]: Sure! `save()` is what triggers our signal handlers, after all 👌

File: docs/querying_history.rst (hunk 4 above)
Comment (ddabble) [reply docs/querying_history.rst:197]: > Maybe we put this after the code example?
> > Historical records will always be created using the ``create`` manager method.

Sure, I can place it inside a `.. note::` :)

File: docs/querying_history.rst (hunk 5 above)
Comment (ddabble) [reply docs/querying_history.rst:219]: Fair, I'll add it :)

File: simple_history/models.py (hunk 6 above)
Comment (ddabble) [reply simple_history/models.py:185]: We do have somewhat similar markup in other docstrings (like double instead of single backticks) that's useful when rendering them with previewing tools in some IDEs 🤷 Though I do see that it doesn't add that much value in this case, and it looks slightly out of place when reading the code directly, so I'll remove it 🙂

File: docs/querying_history.rst (hunk 1 above)
Comment (tim-schilling) [reply docs/querying_history.rst]: I prefer the normal casing for sentence titles here, but it's not something I'd argue to change an existing set of documentation over to.

File: docs/querying_history.rst (hunk 4 above)
Comment (tim-schilling) [reply docs/querying_history.rst:195]: > You think we should do that in this PR?

Definitely no :smile: A context manager sounds like a fantastic idea. Dang, that's clever!

**Review by tim-schilling — APPROVED** [review]
:rocket:

File: docs/querying_history.rst (hunk 4 above)
Comment (ddabble) [reply docs/querying_history.rst:195]: Alright :) I agree, though credits for the idea goes to https://github.com/jazzband/django-simple-history/issues/642#issuecomment-922110914 😄
